packages/core/src/components/essentials/tooltip.ts

```diff
@@ -26,26 +29,45 @@ export class Tooltip extends Component {
 		this.init();
 	}
 
-	init() {
-		// Grab the tooltip element
-		const holder = select(this.services.domUtils.getHolder());
-		const chartprefix = Tools.getProperty(
-			this.model.getOptions(),
-			"style",
-			"prefix"
-		);
-		this.tooltip = DOMUtils.appendOrSelect(
-			holder,
-			`div.${settings.prefix}--${chartprefix}--tooltip`
-		);
+	handleShowTooltip = (e) => {
+		const data = e.detail.data || e.detail.items;
+		const defaultHTML = this.getTooltipHTML(e);
 
-		// Apply html content to the tooltip
-		const tooltipTextContainer = DOMUtils.appendOrSelect(
-			this.tooltip,
-			"div.content-box"
-		);
-		this.tooltip.style("max-width", null);
+		// if there is a provided tooltip HTML function call it
+		if (
+			Tools.getProperty(
+				this.model.getOptions(),
+				"tooltip",
+				"customHTML"
+			)
+		) {
+			if (e.detail.content) {
+						const labelHTML = `<div class="title-tooltip">${e.detail.content}</div>`;
+						this.tooltipTextContainer.html(labelHTML);
```

Comment:
again, `tooltipTextContainer` shouldn't be stored in `this`, there's no point in doing that. If it's really only needed here then we should just move it over here